### PR TITLE
Game init config

### PR DESCRIPTION
**Pull request by invisime**
This enhances and streamlines the options for initializing a game. Decks default to having more sun cards (to be more in line with the "real" game). The `OneCardOfEachColor` helper was moved and made public since it's now part of deck initialization.

### CODE REVIEW

**Review by jdwile — APPROVED** [review]
LGTM :+1:

File: GameEngine/GameOptions.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+{
+    public class GameOptions
+    {
+        public int ColoredCardsPerColor { get; private set; }
+        public int ColoredSpacesPerColor { get; private set; }
+        public int SunCards { get; private set; }
+        public int SunSpaces { get; private set; }
+        public int Owls { get; private set; }
+
+        public static GameOptions FromMultiplier(int gameSizeMultiplier)
```
Comment (rpherbig) [inline GameEngine/GameOptions.cs]: ❓ Should the default constructor be overridden/private?

Comment (rpherbig) [inline GameEngine/GameOptions.cs:13]: ❓ Should the default constructor be overridden and made private?

File: GameEngine/GameState.cs [hunk 2]
```diff
@@ -6,19 +6,22 @@ namespace GameEngine
...
+        public int SunSpaces { get; }
         public int SunCounter { get; private set; }
-        public int GameSizeMultiplier { get; }
-        
-        public GameState(IPlayer player, Deck deck, int gameSizeMultiplier)
+
+        public GameState(int multiplier, params IPlayer[] player)
+            : this(GameOptions.FromMultiplier(multiplier), player) { }
+
+        public GameState(GameOptions options, params IPlayer[] players)
```
Comment (rpherbig) [inline GameEngine/GameState.cs]: ❓ Why does this allow multiple players when only the first is used?

File: GameEngineTests/GameStateTests.cs [hunk 3]
```diff
@@ -18,11 +18,11 @@ public class GameStateTests
...
         public void TestInitialize()
         {
-            Player = new LeastRecentCardPlayer();
             Multiplier = 6;
-            NumberOfOwls = 6;
-            Deck = new Deck(Multiplier);
-            State = new GameState(Player, Deck, Multiplier);
+            Player = new LeastRecentCardPlayer();
+            State = new GameState(Multiplier, Player);
+            Deck = State.Deck;
```
Comment (rpherbig) [inline GameEngineTests/GameStateTests.cs]: 💡 We're a little inconsistent accessing `Deck` vs `State.Deck`. Give this change (of Deck being a part of State), I prefer removing the `Deck` var and using `State.Deck` everywhere. Thoughts?

File: GameEngine/GameOptions.cs (hunk 1 above)
Comment (invisime) [reply GameEngine/GameOptions.cs:13]: Sure. I suspect we'll need to modify it later, but we can do that later.

File: GameEngine/GameState.cs (hunk 2 above)
Comment (invisime) [reply GameEngine/GameState.cs]: Looking ahead to future stories. Removed because YAGNI.

File: GameEngineTests/GameStateTests.cs (hunk 3 above)
Comment (invisime) [reply GameEngineTests/GameStateTests.cs]: Done.

**Pull request merged**